src/services/Tex2Svg.php

```diff
@@ -56,26 +57,61 @@ public function getContent(): string
         // based on https://github.com/mpdf/mpdf-examples/blob/master/MathJaxProcess.php
         $sizeConverter = new SizeConverter($this->mpdf->dpi, $this->mpdf->default_font_size, $this->mpdf, new NullLogger());
 
-        // scale SVG size according to pdf + font settings
-        // only select mathjax svg
-        preg_match_all('/<mjx-container[^>]*><svg([^>]*)/', $html, $mathJaxSvg);
-        foreach ($mathJaxSvg[1] as $svgAttributes) {
-            preg_match('/width="(.*?)"/', $svgAttributes, $wr);
-            preg_match('/height="(.*?)"/', $svgAttributes, $hr);
-
-            if ($wr && $hr) {
-                $w = $sizeConverter->convert($wr[1], 0, $this->mpdf->FontSize) * $this->mpdf->dpi / 25.4;
-                $h = $sizeConverter->convert($hr[1], 0, $this->mpdf->FontSize) * $this->mpdf->dpi / 25.4;
+        // scale SVG size according to pdf + font settings,
+        // convert nested SVGs to PNGs here as mPDF cannot do it, v8.0.13
+        // get all MathJax SVGs, use named subpatterns ('svg', 'attributes'), delimiter is '#'
+        preg_match_all(
+            '#<mjx-container[^>]*>(?P<svg><svg(?P<attributes>[^>]*)>[\s\S]*?)</mjx-container>#',
+            $html,
+            $mathJaxSvgs,
+            PREG_SET_ORDER
+        );
+        foreach ($mathJaxSvgs as $mathJaxSvg) {
+            // get the SVG dimensions
+            preg_match('/width="(?P<value>.*?)"/', $mathJaxSvg['attributes'], $width);
+            preg_match('/height="(?P<value>.*?)"/', $mathJaxSvg['attributes'], $height);
+            $w = $width['value'];
+            $h = $height['value'];
+
+            if ($width && $height) {
+                // MathJax dimensions are in 'ex', convert() returns 'mm' -> final is pixel
+                // scale SVG size according to pdf + font settings
+                $scaleFactor = $this->mpdf->dpi / 25.4;
```

Comment:
what is this magic number 25.4? Use a private const instead at top of class with a clear name.

Reply:
This is 25.4 mm/inch. Will add MM_PER_INCH

Reply:
I think a better name would be: INCH_CONVERSION_FACTOR or similar.